eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/ZmqWatcher.scala

```diff
@@ -229,19 +234,32 @@ private class ZmqWatcher(chainHash: ByteVector32, blockCount: AtomicLong, client
           }
         Behaviors.same
 
-      case ProcessNewBlock(block) =>
-        log.debug("received blockid={}", block.blockId)
+      case ProcessNewBlock(blockHash) =>
+        log.debug("received blockhash={}", blockHash)
         log.debug("scheduling a new task to check on tx confirmations")
+        // we have received a block, so we can reset the block timeout timer
+        timers.startSingleTimer(TickBlockTimeout, TickBlockTimeout, blockTimeout)
         // we do this to avoid herd effects in testing when generating a lots of blocks in a row
         timers.startSingleTimer(TickNewBlock, TickNewBlock, 2 seconds)
         Behaviors.same
 
+      case TickBlockTimeout =>
+        // we haven't received a block in a while, we check whether we're behind and restart the timer.
+        timers.startSingleTimer(TickBlockTimeout, TickBlockTimeout, blockTimeout)
+        val currentBlockCount = blockCount.get()
+        client.getBlockCount.map { count =>
+          if (count > currentBlockCount) {
+            context.log.warn("new block wasn't received via ZMQ, you should verify your bitcoind node")
+            context.self ! TickNewBlock
+          }
+        }
```

Comment:
A simpler logic would have been to just send `TickNewBlock` unconditionnally.

Reply:
That would be more expensive, because we would go through the `checkConfirmed` code for every `WatchConfirmed` for no reason...and I like being able to have a log line for that case where ZMQ seems to be behind, so once we have that `if` it makes sense to use it to trigger work as well :slightly_smiling_face: